locallibrary/catalog/views.py

```diff
@@ -1,3 +1,23 @@
 from django.shortcuts import render
+from .models import Book, Author, BookInstance, Genre
 
-# Create your views here.
+
+def index(request):
+    """View function for home page of site"""
+
+    # Generate counts of the main objects
+    num_books = Book.objects.count()
+    num_instances = BookInstance.objects.count()
+    num_instances_available = BookInstance.objects.filter(status__exact='a').count()# Available books (status = 'a')
```

Comment:
Tìm hiểu về các Fields lookup khác ngoài `exact`

Reply:
iexact: Tương tự như exact, nhưng không phân biệt chữ hoa chữ thường.

contains: Trả về các đối tượng có trường chứa một chuỗi cụ thể.

icontains: Giống như contains, nhưng không phân biệt chữ hoa chữ thường.

startswith: Trả về các đối tượng có trường bắt đầu bằng một chuỗi cụ thể.

istartswith: Giống như startswith, nhưng không phân biệt chữ hoa chữ thường.

endswith: Trả về các đối tượng có trường kết thúc bằng một chuỗi cụ thể.

iendswith: Giống như endswith, nhưng không phân biệt chữ hoa chữ thường.

gt (greater than): Trả về các đối tượng có giá trị trường lớn hơn một giá trị cụ thể.

lt (less than): Trả về các đối tượng có giá trị trường nhỏ hơn một giá trị cụ thể.

gte (greater than or equal to): Trả về các đối tượng có giá trị trường lớn hơn hoặc bằng một giá trị cụ thể.

lte (less than or equal to): Trả về các đối tượng có giá trị trường nhỏ hơn hoặc bằng một giá trị cụ thể.

in: Trả về các đối tượng có giá trị trường nằm trong một danh sách giá trị cụ thể.

range: Trả về các đối tượng có giá trị trường nằm trong một khoảng giá trị cụ thể.